src/queue.c

```diff
@@ -664,6 +669,10 @@ _dispatch_root_queues_init_workq(int *wq_supported)
 			}
 #endif // DISPATCH_USE_LEGACY_WORKQUEUE_FALLBACK
 			qc->dgq_kworkqueue = pwq ? pwq : (void*)(~0ul);
+			// because the fastpath of _dispatch_global_queue_poke didn't
+			// know yet that we're using the internal pool implementation
+			// we have to undo its setting of dgq_pending
+			qc->dgq_pending = 0;
```

Comment:
this is really the only one of my comments that requires fixing, the rest of my comments are more about fitting in better with the existing configuration/define scheme and some style nits that we can also address later if necessary